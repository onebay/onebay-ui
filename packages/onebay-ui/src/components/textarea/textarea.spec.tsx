import { mount } from '@vue/test-utils'
import Textarea from './index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(Textarea, {
    slots,
    props: { value: '', ...values },
  })
}

describe('Textarea Snap', () => {
  it('render initial Textarea', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Textarea -- props style', () => {
    const wrapper = factory({ style: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Textarea -- props className', () => {
    const wrapper = factory({ className: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Textarea -- props value', () => {
    const wrapper = factory({ value: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Textarea -- props maxLength', () => {
    const wrapper = factory({ maxLength: 300 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Textarea -- props placeholder', () => {
    const wrapper = factory({ placeholder: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Textarea -- props count', () => {
    const wrapper = factory({ count: false })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Textarea -- props height', () => {
    const wrapper = factory({ height: 3000 })
    expect(wrapper.vm.height).toBe(3000)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Textarea textOverflowForbidden false', () => {
    const wrapper = factory({ textOverflowForbidden: false })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('Textarea Event', () => {
  it('Textarea onChange Event', () => {
    const wrapper = factory({ value: 'test' })
    wrapper.find('.ob-textarea__textarea').trigger('input')
    expect(wrapper.emitted()).toHaveProperty('change')
  })

  it('Textarea onBlur Event', () => {
    const wrapper = factory({ value: 'test' })
    wrapper.find('.ob-textarea__textarea').trigger('blur')
    expect(wrapper.emitted()).toHaveProperty('blur')
  })

  it('Textarea onFocus Event', () => {
    const wrapper = factory({ value: 'test' })
    wrapper.find('.ob-textarea__textarea').trigger('focus')
    expect(wrapper.emitted()).toHaveProperty('focus')
  })
})
