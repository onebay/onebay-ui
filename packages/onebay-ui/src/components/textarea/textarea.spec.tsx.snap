// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea Snap render Textarea -- props className 1`] = `
<div
  class="ob-textarea ob-textarea--WEB test"
>
  <textarea
    class="ob-textarea__textarea"
    maxlength="200"
    placeholder=""
  />
  <div
    class="ob-textarea__counter"
  >
    0
    /
    200
  </div>
</div>
`;

exports[`Textarea Snap render Textarea -- props count 1`] = `
<div
  class="ob-textarea ob-textarea--WEB"
>
  <textarea
    class="ob-textarea__textarea"
    maxlength="200"
    placeholder=""
  />
  <!---->
</div>
`;

exports[`Textarea Snap render Textarea -- props style 1`] = `
<div
  class="ob-textarea ob-textarea--WEB"
  style="color: red;"
>
  <textarea
    class="ob-textarea__textarea"
    maxlength="200"
    placeholder=""
  />
  <div
    class="ob-textarea__counter"
  >
    0
    /
    200
  </div>
</div>
`;

exports[`Textarea Snap render Textarea -- props height 1`] = `
<div
  class="ob-textarea ob-textarea--WEB"
>
  <textarea
    class="ob-textarea__textarea"
    maxlength="200"
    placeholder=""
    style="height: 3000px;"
  />
  <div
    class="ob-textarea__counter"
  >
    0
    /
    200
  </div>
</div>
`;

exports[`Textarea Snap render Textarea -- props maxLength 1`] = `
<div
  class="ob-textarea ob-textarea--WEB"
>
  <textarea
    class="ob-textarea__textarea"
    maxlength="300"
    placeholder=""
  />
  <div
    class="ob-textarea__counter"
  >
    0
    /
    300
  </div>
</div>
`;

exports[`Textarea Snap render Textarea -- props placeholder 1`] = `
<div
  class="ob-textarea ob-textarea--WEB"
>
  <textarea
    class="ob-textarea__textarea"
    maxlength="200"
    placeholder="test"
  />
  <div
    class="ob-textarea__counter"
  >
    0
    /
    200
  </div>
</div>
`;

exports[`Textarea Snap render Textarea -- props value 1`] = `
<div
  class="ob-textarea ob-textarea--WEB"
>
  <textarea
    class="ob-textarea__textarea"
    maxlength="200"
    placeholder=""
  />
  <div
    class="ob-textarea__counter"
  >
    4
    /
    200
  </div>
</div>
`;

exports[`Textarea Snap render Textarea textOverflowForbidden false 1`] = `
<div
  class="ob-textarea ob-textarea--WEB"
>
  <textarea
    class="ob-textarea__textarea"
    maxlength="700"
    placeholder=""
  />
  <div
    class="ob-textarea__counter"
  >
    0
    /
    200
  </div>
</div>
`;

exports[`Textarea Snap render initial Textarea 1`] = `
<div
  class="ob-textarea ob-textarea--WEB"
>
  <textarea
    class="ob-textarea__textarea"
    maxlength="200"
    placeholder=""
  />
  <div
    class="ob-textarea__counter"
  >
    0
    /
    200
  </div>
</div>
`;
